src/model/wallet.ts

```diff
@@ -720,28 +776,38 @@ export const createWallet = async (
       // Encrypt with the PIN
       const encryptedPkey = await encryptor.encrypt(userPIN, pkey);
       if (encryptedPkey) {
-        await savePrivateKey(walletAddress, encryptedPkey);
+        await savePrivateKey(walletAddress, encryptedPkey, isHardwareWallet);
```

Comment:
same as `loadPrivateKey`  - I wouldn't change this fn at all. 
You can create a new one `saveHardwareWalletKey` and call it instead when `isHardwareWallet=true`